octoprint/datadog_checks/octoprint/check.py

```diff
@@ -0,0 +1,138 @@
+# After installing the Datadog Agent on your OctoPrint device, create an API
+# key from the OctoPrint service, and provide it.
+# Drop this file into `/etc/datadog-agent/checks.d` along with an empty file
+# named `octoprint.yaml` in `/etc/datadog-agent/conf.d`.  Pass in your
+# OctoPrint API key as an instance value like so:
+# ```
+# instances:
+#   - octo_api_key: APIKEYGOESHERE
+# ```
+
+# TODO:
+# - Refactor Octo HTTP API calls
+
+import logging
+import os
+
+import requests
+
+from datadog_checks.base import AgentCheck
+from datadog_checks.base.utils.subprocess_output import get_subprocess_output
+
+__version__ = "0.1.2"
+
+TIMEOUT = 10
+
+def __init__():
+    logger = logging.getLogger('{}.{}'.format(__name__, self.name))
+    self.log = CheckLoggingAdapter(logger, self)
+    hostname = os.popen("hostname").readline().strip()
+    self.log.debug('hostname: %s', hostname)
+    SERVER = "http://" + hostname
+    self.log.debug('OctoSERVER: %s', SERVER)
+
+class OctoPrintCheck(AgentCheck):
```

Comment:
```suggestion
class OctoPrintCheck(AgentCheck):

    def __init__(self, name, init_config, instances):
        super(Apache, self).__init__(name, init_config, instances)
        self.log.debug('OctoPrint monitoring starting on %s', self.hostname)
```

The initializer method here can be used for parsing the config and setting up instance variables.

Reply:
What is the "Apache" on line 37 refer to?

Reply:
Copy/paste error, it should be `OctoPrintCheck` instead.

Reply:
👍